examples/dogsvscats/model_byol.go

```diff
@@ -4,14 +4,14 @@ package dogsvscats
 // https://arxiv.org/abs/2006.07733
 
 import (
-	"github.com/gomlx/exceptions"
+	inceptionv4 "github.com/gomlx/gomlx/examples/inceptionv3"
```

Comment:
why named "v3" v4?

Reply:
Ha, I always read "if" in the end as an "interface" short name, I hadn't realized the mix up. Thank your for the  "builderiface" suggestion, using it instead.

About the reflection: this happens only during the computation graph building  -- this code only tells what operations need to be performed. The results of this is JIT-compiled and later, it is executed without all this reflection -- well there are other reflection needed later, but much less. It only builds and JIT-compiles the graph once, which presumably later will be executed N times ... so the speed during graph building is not a concern ... JIT-compilation takes orders of magnitude more time in the end.